Packs/SymantecEndpointSecurity/Integrations/SymantecEndpointSecurity/SymantecEndpointSecurity.py

```diff
@@ -0,0 +1,459 @@
+            raise DemistoException(
+                f"The key 'access_token' does not exist in response, Response from API: {res}"
+            )
+
+    def get_events(self, payload: dict[str, str]):
+        """
+        API call in streaming to fetch events
+        """
+        return self._http_request(
+            method="POST",
+            url_suffix=f"/v1/event-export/stream/{self.stream_id}/{self.channel_id}",
+            json_data=payload,
+            params={"connectionTimeout": DEFAULT_CONNECTION_TIMEOUT},
+            resp_type="text",
+            headers=self.headers,
+        )
+
+
+def manage_fetch_interval(client, start_run, end_run):
+    """
+    Manages the fetch interval by sleeping if necessary.
+
+    This function calculates the fetch runtime against client.fetch_interval.
+    If the runtime is less than the client.fetch_interval time, it will sleep
+    for the time difference between client.fetch_interval and the fetch runtime.
+    Otherwise, the next fetch will occur immediately.
+
+    Args:
+        client: The client object containing the fetch_interval attribute.
+        start_run: The start time of the fetch operation.
+        end_run: The end time of the fetch operation.
+    """
+    fetch_sleep = client.fetch_interval - (end_run - start_run)
+    if fetch_sleep > 0:
+        time.sleep(fetch_sleep)
+
+
+def normalize_date_format(date_str: str) -> str:
+    """
+    Normalize the given date string by removing milliseconds.
+
+    Args:
+        date_str (str): The input date string to be normalized.
+
+    Returns:
+        str: The normalized date string without milliseconds.
+    """
+    try:
+        # Parse the original date string with milliseconds
+        original_date = datetime.strptime(date_str, "%Y-%m-%dT%H:%M:%S.%fZ")
+    except Exception:
+        if "." in date_str:
+            date_str = f"{date_str.split('.')[0]}Z"
+        return date_str
+
+    # Convert back to the desired format without milliseconds
+    new_date_str = original_date.strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    return new_date_str
+
+
+def update_new_integration_context(
+    filtered_events: list[dict[str, str]],
+    next_hash: str,
+    include_last_fetch_events: bool,
+    last_integration_context: dict[str, str],
+):
+    """
+    Update the integration context.
+
+    Args:
+        filtered_events (list[dict[str, str]]): A list of filtered events.
+        next_hash (str): The hash for the next fetch operation.
+        include_last_fetch_events (bool): Flag to include last fetched events in the integration context.
+        last_integration_context (dict[str, str]): The previous integration context.
+    """
+    events_suspected_duplicates = (
+        extract_events_suspected_duplicates(filtered_events) if filtered_events else []
+    )
+
+    # Determine the latest event time: Extract the last time of the filtered event,
+    # or use the last known time from integration context if no filtered events exist
```

Comment:
move all comments inside their `if`s, to avoid somebody else moving them by mistake